actionpack/lib/action_controller/metal/data_streaming.rb

```diff
@@ -155,7 +156,17 @@ def send_file_headers!(options)
         disposition = options.fetch(:disposition, DEFAULT_SEND_FILE_DISPOSITION)
         unless disposition.nil?
           disposition  = disposition.to_s
-          disposition += %(; filename="#{options[:filename]}") if options[:filename]
+          filename = options[:filename]
+          if filename
+            disposition += %(; filename="#{filename}")
+
+            # Support RFC 6266
+            # http://tools.ietf.org/html/rfc6266#appendix-D
+            unless filename.ascii_only?
+              encoded_filename = CGI::escape(filename)
+              disposition += %(; filename*="#{encoded_filename.encoding}''#{encoded_filename}")
```

Comment:
@jeremy Why are you not escaping spaces? The spec says:

```
value-chars   = *( pct-encoded / attr-char )

...

attr-char     = ALPHA / DIGIT
                   / "!" / "#" / "$" / "&" / "+" / "-" / "."
                   / "^" / "_" / "`" / "|" / "~"
                  ; token except ( "*" / "'" / "%" )
```

Which implies to me that spaces are excluded - they're not in that list and and 2616 specifically excludes them from the definition of "token":

```
      token          = 1*<any CHAR except CTLs or separators>
      separators     = "(" | ")" | "<" | ">" | "@"
                      | "," | ";" | ":" | "\" | <">
                      | "/" | "[" | "]" | "?" | "="
                      | "{" | "}" | SP | HT
```

But if I'm missing something, I'd like to change my implementation too!